src/Fabulous.XamarinForms/Views/Controls/Label.fs

```diff
@@ -71,45 +133,36 @@ type LabelModifiers =
         ) =
-        this.AddScalar(Label.Padding.WithValue(Thickness(left, top, right, bottom)))
+        LabelModifiers.padding (this, Thickness(left, top, right, bottom))
 
     [<Extension>]
-    static member inline lineBreakMode(this: WidgetBuilder<'msg, #ILabel>, value: LineBreakMode) =
-        this.AddScalar(Label.LineBreakMode.WithValue(value))
+    static member inline textColor(this: WidgetBuilder<'msg, #ILabel>, light: Color, ?dark: Color) =
+        this.AddScalar(Label.TextColor.WithValue(AppTheme.create light dark))
 
     [<Extension>]
-    static member inline centerTextHorizontal(this: WidgetBuilder<'msg, #ILabel>) =
-        this.AddScalar(Label.HorizontalTextAlignment.WithValue(TextAlignment.Center))
+    static member inline textDecoration(this: WidgetBuilder<'msg, #ILabel>, value: TextDecorations) =
+        this.AddScalar(Label.TextDecorations.WithValue(value))
 
     [<Extension>]
-    static member inline style(this: WidgetBuilder<'msg, #IView>, style: Style) =
-        this.AddScalar(NavigableElement.Style.WithValue(style))
+    static member inline textTransform(this: WidgetBuilder<'msg, #ILabel>, value: TextTransform) =
+        this.AddScalar(Label.TextTransform.WithValue(value))
 
     [<Extension>]
-    static member inline centerTextVertical(this: WidgetBuilder<'msg, #ILabel>) =
-        this.AddScalar(Label.VerticalTextAlignment.WithValue(TextAlignment.Center))
+    static member inline textType(this: WidgetBuilder<'msg, #ILabel>, value: TextType) =
+        this.AddScalar(Label.TextType.WithValue(value))
 
     [<Extension>]
-    static member inline font
-        (
-            this: WidgetBuilder<'msg, #ILabel>,
-            ?size: double,
-            ?namedSize: NamedSize,
-            ?attributes: FontAttributes
-        ) =
-
-        let mutable res = this
-
-        match size with
-        | None -> ()
-        | Some v -> res <- res.AddScalar(Label.FontSize.WithValue(v))
+    static member inline verticalTextAlignment(this: WidgetBuilder<'msg, #ILabel>, value: TextAlignment) =
+        this.AddScalar(Label.VerticalTextAlignment.WithValue(value))
 
-        match namedSize with
-        | None -> ()
-        | Some v -> res <- res.AddScalar(Label.FontSize.WithValue(Device.GetNamedSize(v, typeof<Label>)))
+    [<Extension>]
+    static member inline centerTextHorizontal(this: WidgetBuilder<'msg, #ILabel>) =
+        this.AddScalar(Label.HorizontalTextAlignment.WithValue(TextAlignment.Center))
 
-        match attributes with
-        | None -> ()
-        | Some v -> res <- res.AddScalar(Label.FontAttributes.WithValue(v))
+    [<Extension>]
+    static member inline centerTextVertical(this: WidgetBuilder<'msg, #ILabel>) =
+        this.AddScalar(Label.VerticalTextAlignment.WithValue(TextAlignment.Center))
 
-        res
+    [<Extension>]
+    static member inline style(this: WidgetBuilder<'msg, #IView>, style: Style) =
+        this.AddScalar(NavigableElement.Style.WithValue(style))
```

Comment:
This one should be inside the `_NavigableElement.fs` file

Reply:
I thought that but I notice that this extension used IView and if I move this extension to _NavigableElement.fs it is not going to be usable as we do not have access to that flexible type ?

Reply:
@TimLariviere  We could define all the interfaces in too level file so we can use it without file ordering restrictions ?

Reply:
I think it should be fine to create the interface type `INavigableElement` according to the inheritance hierarchy of XF, and add the extension method to that interface

Reply:
@TimLariviere INavigableElement has been added inheriting  from IView to make it work with the CollectionBuilder and also updated Button so style can be also apply there .  Let me know if that is what you meant ?  😀

Reply:
It's almost that.
But to respect the hierarchy, it would be better to use the same file ordering: Element -> NavigableElement -> VisualElement -> View

So instead of `type INavigableElement = inherit IView`, it would be `type IView = inherit INavigableElement`.

Reply:
Ok Now should fine following suggested hierarchy . I hate inheritance now ;)